lib/src/features/authentication/view/sign_in/sign_in_screen.dart

```diff
@@ -1,17 +1,44 @@
 import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
 import 'package:go_router/go_router.dart';
 import 'package:google_fonts/google_fonts.dart';
-import 'package:online_course_app/gen/assets.gen.dart';
-import 'package:online_course_app/src/constants/app_sizes.dart';
-import 'package:online_course_app/src/router/app_router.dart';
+import 'package:online_course_app/src/features/authentication/view/sign_in/social_sign_in_controller.dart';
+import '../../../../../gen/assets.gen.dart';
+import '../../../../constants/app_sizes.dart';
+import '../../../../router/app_router.dart';
 
 import '../../../../widgets/common/scaffold.dart';
 
-class SignInScreen extends StatelessWidget {
+class SignInScreen extends ConsumerWidget {
   const SignInScreen({super.key});
 
   @override
-  Widget build(BuildContext context) {
+  Widget build(BuildContext context, WidgetRef ref) {
+    ref.listen(socialSignInControllerProvider, (previous, next) {
```

Comment:
logic (tính toán, navigate, ...) em dời sang controller nha

Reply:
Này là code UI mà anh, với hàm này chỉ chạy trong hàm build thôi @krish-do-goldenowl